pkg/service/store/store.go

```diff
@@ -165,6 +171,135 @@ func NewClusterService(name, namespace string) ClusterService {
 	}
 }
 
+// MCSAPIServiceSpec is similar as ServiceImportSpec except for the fields: IPs,
+// ExportCreationTimestamp and ServiceName.
+//
+// +k8s:deepcopy-gen=true
+type MCSAPIServiceSpec struct {
+	// ServiceName is the name of the ServiceImport or ServiceExport parent resources.
+	ServiceName string `json:"serviceName"`
+	// ExportCreationTimestamp is a timestamp representing the server time when this object was
+	// created. It is not guaranteed to be set in happens-before order across separate operations.
+	// Clients may not set this value. It is represented in RFC3339 form and is in UTC.
+	//
+	// Populated by the system.
+	// Read-only.
+	// Null for lists.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	ExportCreationTimestamp metav1.Time `json:"exportCreationTimestamp"`
+	// +listType=atomic
+	Ports []mcsapiv1alpha1.ServicePort `json:"ports"`
```

Comment:
Embedding k8s objects into our kvstore json-marshalled backwards compatible objects seems a bit scary to me. We also do want to avoid too many k8s specific things at this layer to somewhat keep the path open to alternative systems in the future. It's also hard to judge how well space-optimized this is. @giorio94 I guess you're our kvstore guy. What do you think about how the `ClusterService` format should evolve and what our rules around it should be? My instinct is that we should keep these structs very minimal and very self-contained.

A related note is that I find it fairly scary that our kvstore objects aren't versioned at all. I wonder if we should at some point think about having a proper schema for all kvstore objects that we can validate? Perhaps going via protobuf or CUE...

Reply:
I agree here with Jussi's comment that the kvstore structs should be maintained as simple as possible, and embedding structs that we don't control feels pretty scary. 

It is also not totally clear to me whether we really need all the fields that are introduced here (e.g., the timestamp), and what would be their use-case. Would the idea be to automatically create `ServiceImport` objects in the target clusters out of this data? In that case it may be worth exploring the idea of storing only the information in a separate kvstore path, as we would be deviating from the original purpose of the ClusterService structure (for instance information about port mappings is already present, but laid out in a different format). That way, we would also reduce the risks of introducing backward-incompatible changes affecting existing functionalities. In any case, a CFP-like discussion would really help IMO.

> A related note is that I find it fairly scary that our kvstore objects aren't versioned at all.

Technically speaking, kvstore paths are versioned (and all at `v1`), but we never explored the possibility of supporting multiple versions in parallel till now AFAIK.